bitwarden_license/src/Sso/Controllers/AccountController.cs

```diff
@@ -407,106 +402,106 @@ public async Task<IActionResult> Logout(string logoutId)
             }
 
             OrganizationUser orgUser = null;
-            if (orgId.HasValue)
+            var organization = await _organizationRepository.GetByIdAsync(orgId);
+            if (organization == null)
             {
-                var organization = await _organizationRepository.GetByIdAsync(orgId.Value);
-                if (organization == null)
+                throw new Exception(_i18nService.T("CouldNotFindOrganization", orgId));
+            }
+            
+            // Try to find OrgUser via existing User Id (accepted/confirmed user)
+            if (existingUser != null)
+            {
+                var orgUsersByUserId = await _organizationUserRepository.GetManyByUserAsync(existingUser.Id);
+                orgUser = orgUsersByUserId.SingleOrDefault(u => u.OrganizationId == orgId);
+            }
+
+            // If no Org User found by Existing User Id - search all organization users via email
+            if (orgUser == null)
+            {
+                var orgUsersByOrgId = await _organizationUserRepository.GetManyByOrganizationAsync(orgId, null);
```

Comment:
Ah, okay, this is where we discussed creating a new stored procedure to find an org user by OrgId + Email, vs. reading all the org users and filtering further by email.... not sure I'm going to necessarily be a stickler on that, while performance will take a bit of a hit for larger organization.... most of those organizations aren't going to be going through this "flow" very often, so it won't be high-frequency enough to spend too much time factoring for performance up-front.

Reply:
Sproc created. Conditional updated.